photon-ml/src/integTest/scala/com/linkedin/photon/ml/GameModelGenerator.scala

```diff
@@ -0,0 +1,102 @@
+import com.linkedin.photon.ml.supervised.regression.LinearRegressionModel
+
+/**
+ * A generator of Game models.
+ *
+ * The models generated are random, within certain parameters that the user can specify (e.g., number of random effects,
+ * number of features in each random effect, ...).
+ */
+class GameModelGenerator {
+
+  private val log : Logger = Logger.getLogger(classOf[GameModelGenerator])
+  private val rng = scala.util.Random
+
+  private def randInt(M: Int) = 1 + math.abs(rng.nextInt) % M
+
+  private def randSparseVector(M: Int) = {
+    val indices = (0 until M).map { _ => randInt(M) }.toSet.toArray.sorted
+    val values = indices.indices.map { _ => rng.nextDouble }
+    (indices, values)
+  }
+
+  /**
+   * Generate a random Game model according to parameters configured by the user.
+   *
+   * @param sc The Spark context
+   * @param taskType The type of model (see TaskType for supported types)
+   * @param NRE The number of random effects in the generated model
+   * @param maxNF The maximum number of features in each random effect. This is only a maximum, because the actual
+   *              number will be drawn at random from the uniform distribution over [1, maxNF).
+   * @param maxNSubs The maximum number of sub-models per random effect. This is again only a maximum, the actual
+   *                 number being drawn at random from the uniform distribution over [1, maxNSubs).
+   * @return a well formed Game model.
+   */
+  def generateGameModel(
+      sc: SparkContext,
+      taskType: TaskType,
+      NRE: Int = 100,
+      maxNF: Int = 100,
+      maxNSubs: Int = 100): GAMEModel = {
+
+    log.info(
+      s"Creating a $taskType Game model " +
+        s"with $NRE random effects, " +
+        s"at most $maxNF features per random effect " +
+        s"and at most $maxNSubs sub-models per random effect")
+
+    val numFeatures = (0 until NRE).map { i => (s"Effect$i", randInt(maxNF)) }.toMap
+
+    val model = taskType match {
+      case TaskType.LINEAR_REGRESSION => LinearRegressionModel.apply _
+      case TaskType.LOGISTIC_REGRESSION => LogisticRegressionModel.apply _
+    }
+
+    val models = numFeatures
+      .map {
+        case (effect, nf) =>
+          val (indices, values) = randSparseVector(nf)
+          val nSubs = if (effect == "Effect0") 1 else randInt(maxNSubs)
+          (effect, if (effect == "Effect0") {
+            new FixedEffectModel(sc.broadcast(model(Coefficients(numFeatures(effect))(indices:_*)(values:_*))), effect)
+          } else {
+            val subs = (0 until nSubs).map { _ => model(Coefficients(numFeatures(effect))(indices:_*)(values:_*)) }
+            val names = (0 until nSubs).map { i => s"$effect-item$i" }
+            new RandomEffectModel(sc.parallelize(names zip subs), "RE" + effect, effect)
+          })
+      }.toSeq
+
+    log.info("Done generating Game model")
+
+    GAMEModel(taskType, models:_*)
```

Comment:
This section could use better spacing, IMO. Also: the `NRE` doesn't actually correspond to the # of random effects, since the first effect is treated as a fixed effect. Plus, we know that the first effect is always the one which will become a fixed effect model, so it doesn't make sense to add the conditional check into the `map` call. Possible alternative below?
```
val fixedMaxNumFeatures = randInt(maxNF)
val (fixedIndices, fixedValues) = randSparseVector(fixedMaxNumFeatures)
val fixedEffectModel = new FixedEffectModel(
  sc.broadcast(modelGenFunc(Coefficients(fixedMaxNumFeatures)(fixedIndices: _*)(fixedValues: _*))),
  "fixedEffect")

val models = numFeatures.map { case (effect, nf) =>
  val (indices, values) = randSparseVector(nf)
  val nSubs = randInt(maxNSubs)
  val subs = (0 until nSubs).map { _ => modelGenFunc(Coefficients(nf)(indices: _*)(values: _*)) }
  val names = (0 until nSubs).map { i => s"$effect-item$i" }
  val model = new RandomEffectModel(sc.parallelize(names zip subs), "RE" + effect, effect)

  (effect, model)
}

log.info("Done generating Game model")

GAMEModel(taskType, (models + ("fixedEffect" -> fixedEffectModel)).toSeq :_*)
```